src/rich-log/domain.hh

```diff
@@ -0,0 +1,131 @@
+#pragma once
+
+#include <clean-core/macros.hh>
+
+namespace rlog
+{
+namespace verbosity
+{
+enum type : int
+{
+    /// TRACE is used to provide diagnostic information verbosely and in very small steps
+    /// e.g. logging in each iteration of an algorithm
+    /// per default, trace should be compile-time DISABLED (as it can and will slow down the app)
+    Trace = 0,
+
+    /// DEBUG is used to provide diagnostic information that usually do not matter
+    /// e.g. logging the arguments to an algorithm
+    /// per default, debug should be compile-time ENABLED but runtime DISABLED
+    /// this means all message can be viewed post-mortem
+    Debug = 1,
+
+    /// INFO is used to log interesting information that USERS of your algorithm might want to know about
+    /// e.g. the selected graphics card, filenames for large save operations, UI actions
+    /// per default, this logging is enabled
+    Info = 2,
+
+    /// WARNING is used to log suspicious calls / inputs
+    /// e.g. loading empty files, showing empty meshes, degenerate faces
+    /// while they are not strictly errors, you should minimize false positives (users will start to ignore them)
+    /// per default, this logging is enabled
+    Warning = 3,
+
+    /// ERRROR is used for clear errors and wrong use of APIs
+    /// e.g. trying to load a nonexistant file, providing unsupported argument combinations
+    /// these errors do not crash the program, but are clearly wrong
+    /// per default, this logging is enabled, will break, and will record stacktraces
+    Error = 4,
+
+    /// FATAL is used for errors that will, with a high probability, crash/invalidate the program
+    /// e.g. failure to acquire crucial resources, out of memory
+    /// fatal errors also tend to be unrecoverable with no clear way to fix them
+    /// per default, this logging is enabled, will break, and will record stacktraces
```

Comment:
I have similar feelings here even if it's the red-alert highest possible error class. From a library-author perspective fatal logs are usually info after detecting a fatal issue (like GPU crash recovery info for example), where it wouldn't be useful to break on the fatal-level log itself. From a user perspective it sort of blurs the line between LOG(Fatal, ..) and more rigid stuff like asserts and unreachable.

Reply:
not sure what you mean. should there be no default break_on_log?

Reply:
removed from domains. added to locations (also in a _once version), because that's hard to emulate efficiently with a user logger. more sophisticated stuff can be done in user logger